### PR TITLE
@W-15359680: Adding cart endpoint extension samples

### PR DESCRIPTION
**Pull request by balaji0034**
Sample extension for enforcing quantity constraints on a cart item while adding an item to the cart.
This corresponds to the endpoint /commerce/webstores/${webstoreId}/carts/${activeCartOrId}/cart-items
and is identified by the EPN Commerce_Endpoint_Cart_ItemCollection for registration/mapping.

Sample extension for updating the listPrice of a cart item based on the currencyIsoCode during edit operations.
This corresponds to the endpoint /commerce/webstores/${webstoreId}/carts/${activeCartOrId}/cart-items/${cartItemId}
and is identified by the EPN Commerce_Endpoint_Cart_Item for registration/mapping.

### CODE REVIEW

File: commerce/endpoint/cart/CartItemCollectionExtensionSample.cls-meta.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ApexClass xmlns="http://soap.sforce.com/2006/04/metadata">
+    <apiVersion>62.0</apiVersion>
```
Comment (balaji0034) [inline commerce/endpoint/cart/CartItemCollectionExtensionSample.cls-meta.xml:3]: We created the Apex class using the Salesforce CLI command. This auto-generated file contained the API version 59.0, which we updated to 62.0.

File: commerce/endpoint/cart/CartItemExtensionSample.cls-meta.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ApexClass xmlns="http://soap.sforce.com/2006/04/metadata">
+    <apiVersion>62.0</apiVersion>
```
Comment (balaji0034) [inline commerce/endpoint/cart/CartItemExtensionSample.cls-meta.xml:3]: We created the Apex class using the Salesforce CLI command. This auto-generated file contained the API version 59.0, which we updated to 62.0.

File: commerce/endpoint/cart/CartItemCollectionExtensionSample.cls [hunk 3]
```diff
@@ -0,0 +1,45 @@
+/**
+ * Sample extension for updating the quantity of a cart item while adding an item to the cart.
+ * This corresponds to the endpoint /commerce/webstores/${webstoreId}/carts/${activeCartOrId}/cart-items
+ * and is identified by the EPN Commerce_Endpoint_Cart_ItemCollection for registration/mapping.
+ */
+public with sharing class CartItemCollectionExtensionSample extends ConnectApi.BaseEndpointExtension {
```
Comment (balaji0034) [inline commerce/endpoint/cart/CartItemCollectionExtensionSample.cls:6]: We have tested these samples in our local org and org farm.

**Review by cw112233 — APPROVED** [review]

File: commerce/endpoint/cart/CartItemCollectionExtensionSample.cls (hunk 3 above)
Comment (snilakandan13) [reply commerce/endpoint/cart/CartItemCollectionExtensionSample.cls:6]: Please add the Apex tests, we need 75% code coverage on every Apex file being added?

File: commerce/endpoint/cart/CartItemCollectionExtensionSample.cls [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+        // Check if cartItemInput is not null
+        if (cartItemInput != null) {
+            // Retrieve the quantity from the cart item input
+            String quantity = cartItemInput.getQuantity();
+
+            // Check if quantity is not null
+            if (quantity != null) {
+                // Set the quantity to 5
+                cartItemInput.setQuantity('5');
```
Comment (wei-liu-salesforce) [inline commerce/endpoint/cart/CartItemCollectionExtensionSample.cls:32]: > cartItemInput.setQuantity('5')
you don't want to set it to quantity?

**Comment by snilakandan13** [comment]
> These seem more like test cases than examples. Why would someone explicitly set all quantities to 5? Why would you hard code the currency code in the response? That would make it inconsistent with what is on the server (unless they happened to match).
> 
> I don't think we need absolute real-world examples, but it would be good if it was more than a test case and it may accomplish a business need. A couple thoughts on real examples:
> 
> 1. Instead of hard-coding the quantity, how about enforcing a maximum quantity? Something like quantity = MIN(quantity, 100).
> 2. Instead of hard-coding the currency code, maybe set a value that might be of use to the API caller that is not be available otherwise, like the list price.

Agreed, the beforePost must be validation example(something) that the customer might actually want to use it for

File: commerce/endpoint/cart/CartItemCollectionExtensionSample.cls (hunk 4 above)
Comment (balaji0034) [reply commerce/endpoint/cart/CartItemCollectionExtensionSample.cls]: @wei-liu-salesforce, do you mean to update the quantity to 5 and then use `cartItemInput.setQuantity(quantity);`

This logic is changed now. Could you please check?

**Comment by balaji0034** [comment]
> These seem more like test cases than examples. Why would someone explicitly set all quantities to 5? Why would you hard code the currency code in the response? That would make it inconsistent with what is on the server (unless they happened to match).
> 
> I don't think we need absolute real-world examples, but it would be good if it was more than a test case and it may accomplish a business need. A couple thoughts on real examples:
> 
> 1. Instead of hard-coding the quantity, how about enforcing a maximum quantity? Something like quantity = MIN(quantity, 100).
> 2. Instead of hard-coding the currency code, maybe set a value that might be of use to the API caller that is not be available otherwise, like the list price.

@jnadeausf @snilakandan13 This is done. Please check.

**Review by jnadeausf — APPROVED** [review]

**Review by snilakandan13 — APPROVED** [review]
Looks good to me, except that the tests are missing, The tests will be added in a future PR

**Review by ddiazccrz — APPROVED** [review]

**Pull request merged**